packages/cordova-plugin/src/web.ts

```diff
@@ -1,19 +1,49 @@
 import { require } from "cordova";
-import { ClearWatchOptions, PluginError, Position, PositionOptions } from "./definitions";
+import { ClearWatchOptions, OSGLOCPosition, PluginError, Position, PositionOptions } from "./definitions";
 import { ClearWatchOptionsDefault, PositionOptionsDefault } from "./defaults";
 
 var exec = require('cordova/exec');
 
 function getCurrentPosition(options: PositionOptions, success: (output: Position) => void, error: (error: PluginError) => void): void {
-  options = options || PositionOptionsDefault;
-
-  exec(success, error, 'OSGeolocation', 'getCurrentPosition', [options]);
+  options = { ...PositionOptionsDefault, ...options };
+
+  let convertOnSuccess = (position: OSGLOCPosition) => {
+    let convertedPosition: Position = {
+      coords: {
+        latitude: position.latitude,
+        longitude: position.longitude,
+        altitude: position.altitude,
+        accuracy: position.accuracy,
+        heading: position.heading,
+        speed: position.speed,
+        altitudeAccuracy: position.accuracy
+      },
+      timestamp: position.timestamp,
+    }
+    success(convertedPosition)
+  }
+  exec(convertOnSuccess, error, 'OSGeolocation', 'getCurrentPosition', [options]);
 }
 
-function watchPosition(options: PositionOptions, success: (output: string) => void, error: (error: PluginError) => void): void {
+function watchPosition(options: PositionOptions, watchId: string, success: (output: Position) => void, error: (error: PluginError) => void): void {
   options = options || PositionOptionsDefault;
 
-  exec(success, error, 'OSGeolocation', 'watchPosition', [options]);
+  let convertOnSuccess = (position: OSGLOCPosition) => {
+    let convertedPosition: Position = {
+      coords: {
+        latitude: position.latitude,
+        longitude: position.longitude,
+        altitude: position.altitude,
+        accuracy: position.accuracy,
+        heading: position.heading,
+        speed: position.speed,
+        altitudeAccuracy: position.accuracy
+      },
+      timestamp: position.timestamp,
+    }
+    success(convertedPosition)
+  }
+  exec(convertOnSuccess, error, 'OSGeolocation', 'watchPosition', [options, watchId]);
```

Comment:
The `watchId` being passed here seems a bit lost when looking at the rest of the code (`clearWatch` only needs a single property and we're still encapsulating it into his structure). We should keep things consistent, either by creating a `WatchPositionOptions` structure or reusing `ClearWatchOptions` with a better naming.

Reply:
This comment should probably be added to the PR with the JS code. It was included here for testing purposes but this is context of the other PR. With that said, I agree with you.

Reply:
I can fix it there then